python-package/lightgbm/basic.py

```diff
@@ -4912,6 +4912,37 @@ def refit(
         new_booster._network = self._network
         return new_booster
 
+    def refit_tree_manual(self, tree_id: int, values: np.ndarray) -> "Booster":
```

Comment:
AFAIK `set_leaf_outputs` does things a bit differently and was unsure if it would be proper to have 2 functions, one of which does some work behind the scenes.